### PR TITLE
Cog 466 MDC logs

### PR DESCRIPTION
**Pull request by dmytro-palczewski-osi**
### JIRA Issue Link
- [COG-466: Add incident id to all returned exceptions](https://osi-cwds.atlassian.net/browse/COG-466)

### Technical Description
Add request id and user id to logs. Add request id  (aka incindet_id) to all custom error JSON messages.
Implementation is made according to https://moelholm.com/2016/08/16/spring-boot-enhance-your-logging/

### Tests
- [ ] I have included unit tests 
- [x] I have included integration tests 
- [ ] I have included performance tests 
- [ ] I have included other tests 
- [ ] I have NOT included tests 

### Types of changes

- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Checklist:

- [ ] My code is in a stable state ready to be deployable, but not necessarily complete.
- [ ] I promise on my honor as a CWDS developer to ensure I don't break the build, to check Lint/Rubocop, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

File: src/main/java/gov/ca/cwds/config/api/idm/PerryIdmAdminConfiguration.java [hunk 1]
```diff
@@ -1,25 +1,31 @@
...
+  private LoggingFilter loggingFilter;
+
   @Override
   protected void configure(HttpSecurity http) throws Exception {
     http.antMatcher("/idm/permissions/**")
         .authorizeRequests()
         .anyRequest()
-        .hasAnyAuthority("CARES-admin", "CWS-admin");
+        .hasAnyAuthority("CARES-admin", "CWS-admin")
+        .and().addFilterAfter(loggingFilter, UsernamePasswordAuthenticationFilter.class);
```
Comment (sorych) [inline src/main/java/gov/ca/cwds/config/api/idm/PerryIdmAdminConfiguration.java]: I believe this is not the right place to set this filter.
Actually, this configuration is not used and should be removed.

Comment (dmytro-palczewski-osi) [reply src/main/java/gov/ca/cwds/config/api/idm/PerryIdmAdminConfiguration.java]: But manual tests show that this configuration is used

Comment (dmytro-palczewski-osi) [reply src/main/java/gov/ca/cwds/config/api/idm/PerryIdmAdminConfiguration.java]: What about BaseApiConfiguration?

Comment (sorych) [reply src/main/java/gov/ca/cwds/config/api/idm/PerryIdmAdminConfiguration.java]: it was needed when we had a service to overwrite the permissions list. Now it is not needed.

I think BaseApiConfiguration is the best place

Comment (dmytro-palczewski-osi) [reply src/main/java/gov/ca/cwds/config/api/idm/PerryIdmAdminConfiguration.java]: Filter is moved to BaseApiConfiguration

**Comment by sorych** [comment]
@DmitriyRudenko please review the changes to baseApiConfiguration

**Comment by sorych** [comment]
@DmitriyRudenko Currently we widely use the built-in functionality of Spring test framework to put the needed user into security Context. That's why we don't have any authentication errors in our cases.

Let's consider the testing of Authentication mechanism, including logs, as a separate task (technical debt).

**Pull request merged**